rclcpp/include/rclcpp/subscription_base.hpp

```diff
@@ -569,6 +578,8 @@ class SubscriptionBase : public std::enable_shared_from_this<SubscriptionBase>
   uint64_t intra_process_subscription_id_;
   std::shared_ptr<rclcpp::experimental::SubscriptionIntraProcessBase> subscription_intra_process_;
 
+  const SubscriptionEventCallbacks event_callbacks_;
```

Comment:
This is "duplicating" the callbacks object. 
The base class stores them in this variable, while the derived classes store the `options_` object (which also contains them).

Reply:
This duplication is deliberate, the reason being that since the base class is now storing the event callbacks, there is no guarantee that derived classes will store their options object.

There are also cases where there are no derived classes, in which case the lifetime of the event_callbacks object is not guaranteed to be the same as the pub/sub bases. For those reasons I decided to duplicate them (it shouldn't be too significant either way, since it should be just function pointers).

Reply:
Yes, the memory footprint is going to be very small.
However looking at the code I think that maybe the derived class shouldn't have an `options_` member at all.

But this is probably something that can be addressed in a separate PR